### PR TITLE
Bug 1180666 - Manage Homescreens. r=gasolin

### PR DESCRIPTION
**Pull request by Cwiiis**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=589d6292d0b308bee3f39c9376ea33dafa4293fd

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=649b2dffe06fe192b2fbeda7d569d4c7b5b5bab0

File: apps/settings/elements/root.html [hunk 1]
```diff
@@ -84,7 +84,7 @@ <h2 data-l10n-id="personalization"></h2>
           <a id="menuItem-display" class="menu-item" data-icon="brightness" href="#display" data-l10n-id="display"></a>
         </li>
         <li>
-          <a id="menuItem-homescreen" class="menu-item" data-icon="homescreen" href="#homescreen" data-l10n-id="homescreen"></a>
+          <a id="menuItem-homescreen" class="menu-item" data-icon="homescreen" href="#homescreens" data-l10n-id="home-screens"></a>
```
Comment (gasolin) [inline apps/settings/elements/root.html]: menuItem-homescreen`s`

File: apps/settings/js/panels/homescreen_details/homescreen_details.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+      // uninstalling it.
+      var manifestURL = this._app.manifestURL;
+      var homescreenSetting = settings.createLock().get(MANIFEST_URL_PREF);
+      homescreenSetting.onsuccess = () => {
+        var homeManifestURL = homescreenSetting.result[MANIFEST_URL_PREF];
+        if (homeManifestURL === manifestURL) {
+          var settingObject = {};
+          settingObject[MANIFEST_URL_PREF] = DEFAULT_MANIFEST;
+          homescreenSetting = settings.createLock().set(settingObject);
+          homescreenSetting.onsuccess = () => { uninstallApp(); };
```
Comment (gasolin) [inline apps/settings/js/panels/homescreen_details/homescreen_details.js]: please put uninstallApp(); in separate line

```
{
  uninstallApp();
};
```

File: apps/settings/js/panels/homescreens_list/homescreens_list.js [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+
+  var AppsCache = require('modules/apps_cache');
+  var SettingsListener = require('shared/settings_listener');
+  var SettingsService = require('modules/settings_service');
+  var SettingsCache = require('modules/settings_cache');
+  var ManifestHelper = require('shared/manifest_helper');
+
+  var HomescreensList = function ctor_homescreens_list() {
+    this._apps = [];
+    this.elements = {};
```
Comment (gasolin) [inline apps/settings/js/panels/homescreens_list/homescreens_list.js]: this._elements

File: apps/settings/js/panels/homescreens_list/homescreens_list.js [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+     * @param {Number} index link index.
+     * @private
+     */
+    _listItemBuilder: function hl_listItemBuilder(app, index) {
+      var item = document.createElement('li');
+      item.className = 'pack-split';
+      item.dataset.appIndex = index;
+
+      var radioLabel = document.createElement('label');
+      radioLabel.className = 'pack-radio';
```
Comment (gasolin) [inline apps/settings/js/panels/homescreens_list/homescreens_list.js]: is it still necessary to add this class to show radio button here?

File: apps/settings/js/panels/homescreens_list/homescreens_list.js [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+     * @param {Object} app   app object.
+     * @param {Number} index link index.
+     * @private
+     */
+    _listItemBuilder: function hl_listItemBuilder(app, index) {
+      var item = document.createElement('li');
+      item.className = 'pack-split';
+      item.dataset.appIndex = index;
+
+      var radioLabel = document.createElement('label');
```
Comment (gasolin) [inline apps/settings/js/panels/homescreens_list/homescreens_list.js]: To make it more readable, please add a comment sample template above to demonstrate how the generated code looks like, ex:

```
<li class="pack-split" data-appIndex=1>
  <gaia-switch>
    <label></label>
  </gaia-switch>
</li>
```

File: apps/settings/style/lists.css [hunk 6]
```diff
@@ -277,7 +277,8 @@ ul li label {
 }
 
 ul li > label:not([for]),
-ul li > label.pack-switch {
+ul li > label.pack-switch,
+ul li > label.pack-radio {
```
Comment (gasolin) [inline apps/settings/style/lists.css]: please remove `pack-radio`

File: apps/settings/style/lists.css [hunk 7]
```diff
@@ -300,7 +301,8 @@ ul li > label.pack-switch > input ~ span:after {
   right: 1.5rem;
 }
 
-ul li > label.pack-switch:active {
+ul li > label.pack-switch:active,
+ul li > label.pack-radio:active {
```
Comment (gasolin) [inline apps/settings/style/lists.css]: same here

File: apps/settings/style/lists.css [hunk 8]
```diff
@@ -331,7 +333,8 @@ ul li.pack-split > span {
   background-color: transparent;
 }
 
-ul li.pack-split > label.pack-switch {
+ul li.pack-split > label.pack-switch,
+ul li.pack-split > label.pack-radio {
```
Comment (gasolin) [inline apps/settings/style/lists.css]: same here

**Pull request closed without merging**